### PR TITLE
add executable to shell task

### PR DESCRIPTION
**Pull request by samhita-alla**
## Tracking issue
fixes https://github.com/flyteorg/flyte/issues/4664

## Why are the changes needed?

`executable` enables modification of the shell when using `ShellTask`. by default, subprocess in the `ShellTask` uses `/bin/sh`.

## What changes were proposed in this pull request?

`ShellTask` accepts `executable` as an argument.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flytekit/extras/tasks/shell.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def interpolate(
 T = typing.TypeVar("T")
 
 
-def _run_script(script) -> typing.Tuple[int, str, str]:
+def _run_script(script, executable) -> typing.Tuple[int, str, str]:
     """
```
Comment (kumare3) [inline flytekit/extras/tasks/shell.py:149]: Executable should be optional. As we might want to make it work in default

File: flytekit/extras/tasks/shell.py [hunk 2]
```diff
@@ -157,7 +157,15 @@ def _run_script(script) -> typing.Tuple[int, str, str]:
...
-    process = subprocess.Popen(script, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=0, shell=True, text=True)
+    process = subprocess.Popen(
+        script,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        bufsize=0,
+        shell=True,
+        text=True,
+        executable=executable,
```
Comment (eapolinario) [inline flytekit/extras/tasks/shell.py:167]: can we rename this to `shell`? Also, update the docstring.

Comment (samhita-alla) [reply flytekit/extras/tasks/shell.py]: done!

File: flytekit/extras/tasks/shell.py [hunk 3]
```diff
@@ -179,6 +189,7 @@ def __init__(
         script: typing.Optional[str] = None,
         script_file: typing.Optional[str] = None,
         task_config: T = None,
+        shell: str = "/bin/sh",
```
Comment (eapolinario) [inline flytekit/extras/tasks/shell.py:192]: add to docstring.

File: flytekit/extras/tasks/shell.py [hunk 4]
```diff
@@ -145,7 +145,7 @@ def interpolate(
 T = typing.TypeVar("T")
 
 
-def _run_script(script) -> typing.Tuple[int, str, str]:
+def _run_script(script, shell) -> typing.Tuple[int, str, str]:
```
Comment (eapolinario) [inline flytekit/extras/tasks/shell.py:148]: any reason not to type this function?

**Review by eapolinario — APPROVED** [review]

**Pull request merged**